linmod/utils.py

```diff
@@ -1,9 +1,27 @@
 import sys
 from functools import reduce
+from pathlib import Path
 
 import polars as pl
 
 
+class ValidPath(Path):
+    """
+    Functions like `pathlib.Path`, but ensures that the path exists.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.is_dir():
+            self.mkdir(parents=True, exist_ok=True)
```

Comment:
I think this is my most substantial question: is this the behavior you want, that you *always* make the folder? 

Like, you could do something where you check that path already exists and error, unless you pass `ValidPath(mkdir_if_doesnt_exist=True)`

Reply:
I could see this going either way. Right now, re-running the script will just replace the outputs with fresh ones, but we could require some kind of `make clean` if we felt that was better?

Reply:
I do like the idea of a `force=` (or `overwrite`) option. And then maybe in your Makefile you would call `my_script.py --force`, then that `args.force` would propagate to `ValidPath(whatever, force=True)`

You'd want to think about whether `ValidPath(existing_dir)` does nothing or deletes the content of the directory...